src/parser/map_init.c

```diff
@@ -0,0 +1,131 @@
+#include "../../include/cube.h"
+
+void	set_char(int i, char *tmp, int max_length, t_map *m)
+{
+	int			j;
+	int			nl;
+	int			len;
+
+	j = 0;
+	nl = 0;
+	len = ft_strlen(tmp);
+	while (j < len)
+	{
+		m->map[i][j] = tmp[j];
+		j++;
+	}
+	if (tmp[j++] == '\n')
+		nl = 1;
+	else
+	{
+		m->map[i][j] = tmp[j];
+		j++;
+	}
+	if (nl != 0)
+		max_length--;
+	while (j < max_length)
+		m->map[i][j++] = ' ';
+	if (nl != 0)
+		m->map[i][j] = '\n';
```

Comment:
do we need to write the `\n` to the row array?

Reply:
"The map must be parsed as it looks in the file" is written in the subject. I don't know how to interpret this in regard of nl. The task is more into the direction of spaces, at least it seems like in the subject: "Spaces are a valid part of the map and are up to you to handle. You must be able to parse any kind of map, as long as it respects the rules of the map."